crates/bevy_sprite/src/render/mod.rs

```diff
@@ -874,3 +890,96 @@ impl<P: PhaseItem> RenderCommand<P> for DrawSpriteBatch {
         RenderCommandResult::Success
     }
 }
+
+/// Scales a texture to fit within a given quad size with keeping the aspect ratio.
+fn scale_texture(
```

Comment:
I think this makes more sense as a public method on `TextureScale`. This is generally useful logic, and I think there's a good chance we move this out of `bevy_sprite` in the future.

Reply:
`apply_scaling` feels like the right name to me.